include/libdnf/base/transaction.hpp

```diff
@@ -130,6 +135,29 @@ class Transaction {
     /// Retrieve list of problems that occurred during transaction run attempt
     std::vector<std::string> get_transaction_problems() const noexcept;
 
+    /// @brief Check signatures of packages in the resolved transaction.
+    ///
+    /// @param import_keys When the parameter is on, problems related to missing GPG keys are
+    /// tried to be automatically recovered by importing the missing ones.
+    ///
+    /// @return True if all packages have correct signatures or checking is turned off with `gpgcheck` option,
+    /// otherwise false. More info about occurred problems can be retrieved using the `get_gpg_signature_problems`
+    /// method.
+    bool check_gpg_signatures(bool import_keys = true);
+
+    /// @brief Check signatures of packages in the resolved transaction.
+    ///
+    /// @param import_confirm_func This function is used for deciding whether to import the given GPG key
+    /// when a problem related to missing keys occurs.
+    ///
+    /// @return True if all packages have correct signatures or checking is turned off with `gpgcheck` option,
+    /// otherwise false. More info about occurred problems can be retrieved using the `get_gpg_signature_problems`
+    /// method.
+    bool check_gpg_signatures(std::function<bool(const libdnf::rpm::KeyInfo &)> & import_confirm_func);
```

Comment:
`std::function` on public API? Probably a new complication for SWIG binding. Especially if we want to be able to define a callback in the target language.

But there is a solution. The target language can override virtual methods of a wrapped C++ class. Code can 'call up' from C++ into the target language by simply calling a virtual method overridden in a derived class in the target language.

And we use this approach. `libdnf::repo::DownloadCallbacks` for "download'' callbacks and `libdnf::repo::RepoCallbacks` for "import repo keys" callbacks. Previously, the `RepoCallbacks` class was also used for download repository metadata callbacks. I consolidated the download callbacks and today the `RepoCallbacks` class is only used for "import repo key" callback (virtual method `repokey_import`. I wanted to rename the class (to `RepoKeyImportCallbacks` for example).

In "dnf5/context.cpp", the `user_confirm_key` function is basically a copy of the `KeyImportRepoCB::user_confirm_key` method.
**So, the solution is to unify the code and use the class method everywhere.**

I am considering whether to merge this PR as it is. It's a step forward. And perform the unification in the next PR.

Reply:
I know about that and it wasn't intended to be used from SWIG bindings. It is there basically just because of that `Context` usage. I was considering a `friend class` and private method for this purpose, but the `Context` class is outside the library scope, so I didn't think of this as a good idea.

Your proposal looks good, if it's OK, I will do it in another PR and put a TODO comment there for now.

Reply:
@jan-kolarik  OK. I will merge it. A TODO comment is enough for now. You can continue in the next PR.